SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -888,19 +888,22 @@ bool USpatialActorChannel::TryResolveActor()
 	return true;
 }
 
-FObjectReplicator& USpatialActorChannel::PreReceiveSpatialUpdate(UObject* TargetObject)
+FObjectReplicator* USpatialActorChannel::PreReceiveSpatialUpdate(UObject* TargetObject)
 {
+	// If there is no NetGUID for this object, we will crash in FObjectReplicator::StartReplicating, so we verify this here.
 	FNetworkGUID ObjectNetGUID = Connection->Driver->GuidCache->GetOrAssignNetGUID(TargetObject);
 	if (ObjectNetGUID.IsDefault() || !ObjectNetGUID.IsValid())
 	{
-		UE_LOG(LogSpatialActorChannel, Warning, TEXT("PreReceiveSpatialUpdate: NetGUID is invalid! Object: %s"), *TargetObject->GetPathName());
+		// Something went wrong when resolving this object in the PackageMap.
```

Comment:
```suggestion
		// SpatialReceiver tried to resolve this object in the PackageMap, but it didn't propagate to GuidCache.
		// This could happen if the UnrealObjectRef was already mapped to a different object that's been destroyed.
```